packages/twenty-server/src/engine/core-modules/admin-panel/admin-panel.service.ts

```diff
@@ -32,11 +36,32 @@ export class AdminPanelService {
     private readonly loginTokenService: LoginTokenService,
     private readonly twentyConfigService: TwentyConfigService,
     private readonly domainManagerService: DomainManagerService,
+    private readonly auditService: AuditService,
+    private readonly logger: LoggerService,
     @InjectRepository(User)
     private readonly userRepository: Repository<User>,
   ) {}
 
-  async impersonate(userId: string, workspaceId: string) {
+  async impersonate(
+    userId: string,
+    workspaceId: string,
+    impersonatorUserId: string,
+  ) {
+    try {
+      z.string().uuid().parse(userId);
+      z.string().uuid().parse(workspaceId);
+      z.string().uuid().parse(impersonatorUserId);
+    } catch {
+      throw new AuthException('Invalid input', AuthExceptionCode.INVALID_INPUT);
+    }
+
+    const correlationId = randomUUID();
```

Comment:
I feel like this implementation is too local, we should rather tackle this at the request level or somewhere else, because it's a common problem. I'll remove it